src/server/actions/item/create/action.ts

```diff
@@ -0,0 +1,21 @@
+"use server";
+
+import { db } from "@/db";
+import { items, NewItem } from "@/db/schema";
+import { type ItemPostParams } from "@/lib/types";
+
+export async function createItem(itemData: ItemPostParams) {
+  const { image, itemName, itemDescription, itemDate, islost } = itemData;
+
+  const item: NewItem = {
+    image: image,
+    name: itemName,
+    description: itemDescription,
+    date: itemDate,
+    islost: islost,
+  };
+
+  const [newItem] = await db.insert(items).values(item).returning();
```

Comment:
Add an error check, where you return an error in the scenario this case fails please
-> This could fail when drizzle has some problem
-> Input validation fails
-> DB is down and so on!!
```suggestion
  const [newItem] = await db.insert(items).values(item).returning();
```

Also this will be useful
[here](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#server-side-form-validation)
Where we can useActionState for when we do actually build the component, to also mitigate from the user spamming the submit button(potenitally ddos) and also an optimistic update!! (even if it is unlikely haha)

Reply:
ohhh ok yea that would be nice! 

would you recommend changing the form to action={}? A minor change but I think we would have to convert the FormData type that's passed to the action to the NewItem type. Let me know if I'm misunderstanding anything!